<launch>

    <arg name="markerId"        default="5"/>
    <arg name="markerSize"      default="0.5"/>    <!-- in m -->
    <arg name="eye"             default="left"/>
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="ref_frame"       default="base_link"/>  <!-- default="ardrone_base_link"/>  -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->


    <node pkg="aruco_ros" type="single" name="aruco_single" output="screen">
        <remap from="/camera_info" to="/ardrone/bottom/camera_info" />   <!-- /ardrone/bottom/camera_info -->
        <remap from="/image" to="/ardrone/bottom/image_raw" />   <!-- /ardrone/bottom/image_raw -->
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->

<!--        <param name="camera_frame"       value="stereo_gazebo_$(arg eye)_camera_optical_frame"/>  -->
        <param name="camera_frame"       value="ardrone_base_bottomcam"/> <!-- value="ardrone_base_bottomcam"/> -->
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

    <node pkg="rosservice" type="rosservice" name="global_loc" args="call /ardrone/setcamchannel 'channel: 1'  --wait"  output="screen"/>
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen" args="--wait image:=/aruco_single/result"/>
    <node pkg="rostopic" type="rostopic" name="takeoff" args="pub /ardrone/takeoff std_msgs/Empty '{}' " output="screen"/>
<!--     <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop" output="screen"/> -->
    
</launch>

